spacy/tests/doc/test_span_group.py

```diff
@@ -0,0 +1,576 @@
+import pytest
+from random import Random
+from spacy.matcher import Matcher
+from spacy.tokens import Span, SpanGroup
+from ..util import span_lists_equal
+from spacy.tokens.span_group import concat_span_groups
+
+
+@pytest.fixture
+def doc(en_tokenizer):
+    doc = en_tokenizer("0 1 2 3 4 5 6")
+    matcher = Matcher(en_tokenizer.vocab, validate=True)
+
+    matcher.add("4", [[{}, {}, {}, {}]])
+    matcher.add(
+        "2",
+        [
+            [
+                {},
+                {},
+            ]
+        ],
+    )
+    matcher.add(
+        "1",
+        [
+            [
+                {},
+            ]
+        ],
+    )
+
```

Comment:
```suggestion
# fmt: off
    matcher.add("4", [[{}, {}, {}, {}]])
    matcher.add("2", [[{}, {}, ]])
    matcher.add("1", [[{}, ]])
# fmt: on
```
Sorry, I should have also explained that sometimes we "cheat" when the linebreaks intervene with readibility ;-)